Implements infinite scroll for unit transactions

alfonsorNolasco opened the pull request:
Implements infinite scroll for unit transactions.
It also introduces a new API endpoint to retrieve tags for devices.

ARNolasco committed:
Implements infinite scroll for unit transactions

Adds pagination to the unit transaction list, enabling infinite scrolling and improving performance for units with large transaction histories.

Updates the unit service to accept pagination parameters for fetching transactions.
Adds total and transactions properties to the Transaction model

Fixes #8

ARNolasco committed:
Implements infinite scroll for unit transactions

Adds pagination to the unit transaction list, enabling infinite scrolling and improving performance for units with large transaction histories.

Updates the unit service to accept pagination parameters for fetching transactions.
Adds total and transactions properties to the Transaction model

Fixes #8

ARNolasco committed:
Merge remote-tracking branch 'origin/feature/8-infinity-scroll-unit-view' into feature/8-infinity-scroll-unit-view

jda1586 commented on src/app/core/service/unit.service.ts:
```diff
@@ -15,10 +15,13 @@ export class UnitService {
+    })
+  }
+
+  getTags(condoId: string) {
```
shuld be "getDevicesByCondo" or similar

jda1586 replied:
also move this method to CondoService

jda1586 reviewed (changes requested):
pending

ARNolasco committed:
Refactors device retrieval for condo admin

Moves device retrieval logic to the condominium service for better organization and reusability.
Removes the unused `getTags` method from the unit service.
Updates the condo devices component to use the new condominium service method.
Simplifies page update logic in the admin unit component by directly calling `loadTransactions` in `changePage` function.

Related to #8

jda1586 reviewed (approved):
Great!
